src/lustre/lustreNodeGen.ml

```diff
@@ -2096,6 +2118,26 @@ and compile_node_decl gids is_function cstate ctx i ext inputs outputs locals it
     in
     assumes, guarantees, props
   (* ****************************************************************** *)
+  (* Generate Contract Constraints for Array constraints                *)
+  (* ****************************************************************** *)
+  in let (assumes, guarantees, props) =
+    let create_constraint_name rexpr = 
+      Format.asprintf "@[<h>%a@]" A.pp_print_expr rexpr
+    in
+    let over_array_constraints (a, ac, g, gc, p) (pos, id, rexpr, prop_desc) =
+      let sv = H.find !map.state_var (mk_ident id) in
+      let name = create_constraint_name rexpr in
+      let name = name ^ " (" ^ prop_desc ^ ")" in
+      let src = Property.Generated (Some pos, [sv]) in
+      a, ac, g, gc, (sv, name, src, Property.Invariant) :: p
+    in
+    let (assumes, _, guarantees, _, props) = 
+      List.fold_left over_array_constraints
+      (assumes, List.length assumes, guarantees, List.length guarantees, props)
+      gids.GI.array_constraints
+    in
+    assumes, guarantees, props
+  (* ****************************************************************** *)
```

Comment:
I think only `props` is updated, not `assumes` or `guarantees`. You should simplify the code.

Reply:
Addressed